### ISSUE TITLE
Scan stats are misleading: they count files + dirs as files and do not match find

### ISSUE THREAD

**Issue by pombredanne** [nexB/scancode-toolkit#598]
`Scan statistics: 50352 files scanned in 2832s.`
yet
```
$ find . | wc -l
50490
$ find . -type f | wc -l
47434
$ find . -type d | wc -l
3048
$ find . -type l | wc -l
8
$ find .  | grep "\.gitignore"  | wc -l
129
```
The point here is that scancode ignores some files and ignore links and mixes files and dirs in its total.
We should have a way to cross-check that with `find` or be more explicit about what is scanned and not scanned. 

Here we have scancode reporting files+dirs, ignoring links and ignored files (and the  root dir). 
`scancode's 50352 == 50490 - 8 -129-1`
So things add up, but are not clearly explained.

**Comment by armijnhemel** [comment]
Also do not forget pipes, sockets, device nodes, and so on. From a licensing perspective these make no sense to scan. I would probably only look at files, filter everything that is not a file (and for directories: don't follow the symbolic links).

**Comment by pombredanne** [comment]
Every special files and links is ignored alright. Links are never followed. The end-of-scan stats could be made clearer though.

**Comment by pombredanne** [comment]
This has been fixed with #885
